tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.cs

```diff
@@ -396,8 +397,22 @@ private Expression HandleReturnType(Expression finalExpr)
 
         if (typeof(T) != typeof(string))
         {
-            // let the caller throw the correct exception
-            return finalExpr;
+            if (typeof(T) == typeof(double) /* metric probe */
```

Comment:
This seems a bit too explicit - can we make this more explicit or even derive a sub-class that deals with the Metric Probe specific logic?

Reply:
This will make sense?

```
if (typeof(T).IsAssignableFrom(finalExpr.Type))
{
    // return as is
}

if (typeof(T) == typeof(string))
{
    // handle ToString() 
}

if (typeof(T).IsNumeric)
{
    // Try convert to double
}
```

Reply:
> Why not doing it in higher layers - such as when we receive the evaluated result of the metric? (in `ProbeProcessor`)

We can't because the delegate creation will throw (unless we will return `object`)

Reply:
Changed in 1b2566d1a442025260e7614b6c0f95f94133d32f